File: js/src/quick_sort/quick_sort.ts
export function quick_sort(arr: number[]): number[] {
  if (arr.length < 2) {
    return arr;
  }

  const pivot = arr[0];
  const less = [];
  const more = [];

  for (let i = 1; i < arr.length; i++) {
    const element = arr[i];

    if (element <= pivot) {
      less.push(element);
    } else {
      more.push(element);
    }
  }

  return [...quick_sort(less), pivot, ...quick_sort(more)];
}

File: go/quck_sort/quick_sort.go
package quick_sort

func QuickSort(arr []int) []int {
	if len(arr) < 2 {
		return arr
	}

	pivot := arr[0]
	less := []int{}
	more := []int{}

	for _, item := range arr[1:] {
		if item <= pivot {
			less = append(less, item)
		} else {
			more = append(more, item)
		}
	}

	return append(append(QuickSort(less), pivot), QuickSort(more)...)
}
